src/core/server/saved_objects/service/saved_objects_client.ts

```diff
@@ -492,6 +499,8 @@ export class SavedObjectsClient {
           ...object.attributes,
         },
         {
+          ...options,
+          workspaces: [],
```

Comment:
`workspaces: [],` this is not needed because `options.workspaces` won't exist?

Reply:
Yes. From the type definition, options could also contain workspace field. I'm intended to add this because this will ensure the empty workspace to make the function completed and make it looking clear.

Reply:
Can we try to get the rest of params within options without workspaces? There should not have any objects whose workspaces field is empty array.

Reply:
What unexpected errors will be led to? I suppose empty workspaces array just represent that this object doesn't belong to any workspaces and I think we should not parse the options any more if we add a new field.

Reply:
If empty workspace array would be not reliable, I think we can also pass undefined to redeclare. This could ensure rewrite raw option and would be set value by repository.

Reply:
@raintygao You're right, user could set `options.workspaces`. It's safe to make sure we don't send `workspaces` to `this.create`
What about doing this:
```typescript
  deleteFromWorkspaces = async <T = unknown>(
    type: string,
    id: string,
    workspaces: string[],
    options: Omit<SavedObjectsBaseOptions, 'workspaces'> = {}
  ) => {
    if (!workspaces || workspaces.length === 0) {
      throw new TypeError(`Workspaces is required.`);
    }
    if ('workspaces' in options && options.workspaces) {
      throw new TypeError('Invalid options, options.workspaces should not exist.');
    }
```

Reply:
Sure, updated.

Reply:
I remember we use create with overwrite when updating objects with empty workspaces to prevent objects with `workspaces: []` from creating. I could not remember in which case the listing logic may find unexpected objects but we'd better keep the logic as it was before the change right? Or we do not need to use create with overwrite, just update with workspaces: [].

Reply:
Yes, I remember that update object with empty workspace would have some unexpected behaivor. Overwrite create with workspace: undefined would work well, but we can also throw error when options contain workspace field like what we have built for now.

Reply:
Yes, what @ruanyl  proposed should work as well, thanks for making it consistent.